From f70c80894e584cec0894ac2d8086f1ce3add5e94 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 08 Dec 2021 09:06:34 +0200
Subject: [PATCH] [SW-63186] gaudi2: security: set axprot for rot

Set axprot for rotator ifs as per values provided by the vlsi team.

Change-Id: Icef1eb0775cec2ca666145567bfce3a7707b4b1f
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 3de32e2..674f6c6 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -33,6 +33,8 @@
 #include <pcie_wrap_masks.h>
 #include <dcore0_vdec0_mstr_if_core_hbw_regs.h>
 #include <dcore0_vdec0_mstr_if_core_lbw_regs.h>
+#include <rot0_mstr_if_core_hbw_regs.h>
+#include <rot0_mstr_if_core_lbw_regs.h>
 
 /*
  * instances_num : Number of instances of RR to program.
@@ -194,6 +196,7 @@
 
 enum preboot_sec_axprot_regs {
 	AXPROT_VDEC,
+	AXPROT_ROT,
 };
 
 #define DCORE_DIST	0x200000
@@ -915,6 +918,9 @@
 #define AXPROT_SEC_VAL		((PROT_OVR_SECURE << 1) | AXPROT_OVRD_EN)
 #define VDEC_RTR_INST_NUM	2
 #define VDEC_RTR_INST_DIST	0x10000
+#define ROT_RTR_INST_NUM	1
+#define ROT_INST_NUM		2
+#define ROT_INST_DIST		0x10000
 
 static struct axprot_regs axprot_reg[] = {
 	[AXPROT_VDEC] = {
@@ -924,6 +930,14 @@
 		.dcore_num = NUM_OF_DCORES,
 		.inst_dist = VDEC_RTR_INST_DIST,
 		.inst_num = VDEC_RTR_INST_NUM,
+	},
+	[AXPROT_ROT] = {
+		.base_reg = mmROT0_MSTR_IF_CORE_HBW_AXPROT_OVRD,
+		.value = AXPROT_USER_VAL,
+		.dcore_dist = ROT_INST_DIST,
+		.dcore_num = ROT_INST_NUM,
+		.inst_dist = 0,
+		.inst_num = ROT_RTR_INST_NUM,
 	},
 };
 
@@ -953,6 +967,17 @@
 	preboot_axprot_regs_init(AXPROT_VDEC);
 }
 
+static void preboot_secure_rot(void)
+{
+	/* HBW AXPROT */
+	preboot_axprot_regs_init(AXPROT_ROT);
+
+	axprot_reg[AXPROT_ROT].base_reg = mmROT0_MSTR_IF_CORE_LBW_AXPROT_OVRD;
+
+	/* LBW AXPROT */
+	preboot_axprot_regs_init(AXPROT_ROT);
+}
+
 void preboot_security_init(void)
 {
 	preboot_secure_pcie();
@@ -962,6 +987,7 @@
 		preboot_lbw_rtr_preconfig();
 		hl_dev_prot_bits_init();
 		preboot_secure_vdec();
+		preboot_secure_rot();
 		preboot_secure_lbw();
 		preboot_secure_hbw();
 		preboot_puts("Security enabled.\n");
